### PR TITLE
feat(config): make arrangement backfill default

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR makes arrangement backfill the default backfill by changing the `streaming_use_arrangement_backfill` to true by default.

We also fix some ci issues which block this from being the default.

### Changes

Large number of changes are due to `./risedev dapt` changing the planner tests, since now `ArrangementBackfill` will be the scan type instead of `Backfill`.

In this PR we also hide some compactor `table_ids`, since these will take up a lot of logging space and are not really useful in debugging CI.

Here are the regression in PR runtimes:
- end-to-end test 19min -> 21min
- end-to-end test parallel 10min -> 12min
- end-to-end test deterministic sim 21min -> 25min (Could be just regression from other pipelines..)

It's fine IMO since it's just debug builds. More importantly `main-cron` does not show regression for e2e test release or backfill tests.

Here are the changes to main-cron runtimes (see https://github.com/risingwavelabs/risingwave/pull/14846#issuecomment-2011334852 for details):
1. Backfill test seem to improve.
2. e2e release test seem to improve.
3. Recovery test runtimes seem to slightly increase. Perhaps due to the following overheads:
    1. Instead of just using storage table, we now use `ReplicatedStateTable`. Metadata is larger since we need the full `TableCatalog`, instead of just `TableDesc`.
    2. On recovery, we need to read backfill state and recover the vnode level state.

Performance comparisons between this and no shuffle backfill:
- https://github.com/risingwavelabs/risingwave/pull/14846#issuecomment-2002832562

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by kwannoel** [comment]
Need to figure out:
1. Why e2e test will TLE. (Perhaps it's the `get_row` each time we persist state in debug mode).
2. Why deterministic test `test_high_barrier_latency_cancel` takes a long time.

**Comment by kwannoel** [comment]
```
key: e2e_test/streaming/aggregate/hdr_approx_percentile.slt, diff: +1409, arrangement runtime: 10965, no shuffle runtime: 9556

key: e2e_test/streaming/./nexmark/create_views.slt.part, diff: +4248, arrangement runtime: 19953, no shuffle runtime: 15705

key: e2e_test/streaming/nexmark_snapshot.slt, diff: +5114, arrangement runtime: 22997, no shuffle runtime: 17883

key: e2e_test/streaming/nexmark_upstream.slt, diff: +5226, arrangement runtime: 23449, no shuffle runtime: 18223

key: e2e_test/streaming/over_window/main.slt, diff: +2053, arrangement runtime: 49867, no shuffle runtime: 47814

key: e2e_test/streaming/./tpch/./views/q5.slt.part, diff: +1287, arrangement runtime: 2045, no shuffle runtime: 758

key: e2e_test/streaming/./tpch/./views/q7.slt.part, diff: +1782, arrangement runtime: 2563, no shuffle runtime: 781

key: e2e_test/streaming/./tpch/./views/q8.slt.part, diff: +2189, arrangement runtime: 3556, no shuffle runtime: 1367

key: e2e_test/streaming/./tpch/./views/q9.slt.part, diff: +1991, arrangement runtime: 3312, no shuffle runtime: 1321

key: e2e_test/streaming/./tpch/./views/q10.slt.part, diff: +1858, arrangement runtime: 2875, no shuffle runtime: 1017

key: e2e_test/streaming/./tpch/./views/q11.slt.part, diff: +1844, arrangement runtime: 2828, no shuffle runtime: 984

key: e2e_test/streaming/./tpch/./views/q12.slt.part, diff: +1585, arrangement runtime: 2285, no shuffle runtime: 700

key: e2e_test/streaming/./tpch/./views/q13.slt.part, diff: +1560, arrangement runtime: 2252, no shuffle runtime: 692

key: e2e_test/streaming/./tpch/./views/q14.slt.part, diff: +1561, arrangement runtime: 2263, no shuffle runtime: 702

key: e2e_test/streaming/./tpch/./views/q15.slt.part, diff: +1864, arrangement runtime: 2614, no shuffle runtime: 750

key: e2e_test/streaming/./tpch/./views/q16.slt.part, diff: +2220, arrangement runtime: 3210, no shuffle runtime: 990

key: e2e_test/streaming/./tpch/./views/q17.slt.part, diff: +2165, arrangement runtime: 3464, no shuffle runtime: 1299

key: e2e_test/streaming/./tpch/./views/q18.slt.part, diff: +2399, arrangement runtime: 3713, no shuffle runtime: 1314

key: e2e_test/streaming/./tpch/./views/q19.slt.part, diff: +2096, arrangement runtime: 3086, no shuffle runtime: 990

key: e2e_test/streaming/./tpch/./views/q20.slt.part, diff: +3577, arrangement runtime: 5405, no shuffle runtime: 1828

key: e2e_test/streaming/./tpch/./views/q21.slt.part, diff: +3933, arrangement runtime: 5887, no shuffle runtime: 1954

key: e2e_test/streaming/./tpch/./views/q22.slt.part, diff: +2721, arrangement runtime: 4033, no shuffle runtime: 1312

key: e2e_test/streaming/./tpch/create_views.slt.part, diff: +38380, arrangement runtime: 60467, no shuffle runtime: 22087

key: e2e_test/streaming/./tpch/drop_views.slt.part, diff: +4169, arrangement runtime: 8156, no shuffle runtime: 3987

key: e2e_test/streaming/tpch_snapshot.slt, diff: +45753, arrangement runtime: 77609, no shuffle runtime: 31856

key: e2e_test/streaming/tpch_upstream.slt, diff: +44402, arrangement runtime: 77006, no shuffle runtime: 32604

```

Runtime comparison

**Comment by kwannoel** [comment]
Screenshot 2024-01-30 at 5 42 44 PM

Runtime still too long without the debug.

**Comment by kwannoel** [comment]
At least main-cron is not taking a long time for e2e test streaming. Seems to be an issue specific to debug mode.
Screenshot 2024-01-30 at 6 07 51 PM
Screenshot 2024-01-30 at 6 17 25 PM

**Comment by kwannoel** [comment]
Screenshot 2024-03-14 at 2 39 10 PM

- https://buildkite.com/risingwave-test/backfill/builds/387
- https://buildkite.com/risingwave-test/backfill/builds/386

Arrangement backfill passes tests.

**Comment by kwannoel** [comment]
with tombstone:
```
create_watermark_mv_latency(ms): 124532.255
create_mv_latency(ms): 110062.274
batch_query_latency(ms): 28296.241
mv_query_latency(ms): 22080.314
```

without tombstone:
```
create_watermark_mv_latency(ms): 124186.916
create_mv_latency(ms): 114223.373
batch_query_latency(ms): 26236.617
mv_query_latency(ms): 21391.008
```

**Comment by kwannoel** [comment]
To revert this PR:
1. set `streaming_use_arrangement_backfill` back to false by default.
2. Run `./risedev dapt` to reset stream scan to use backfill rather than arrangement backfill.

**Comment by kwannoel** [comment]
Main cron today https://buildkite.com/risingwavelabs/main-cron/builds/2104 seems to be failing as well due to TLE:
Screenshot 2024-03-21 at 2 52 11 PM


For this main-cron for this PR: https://buildkite.com/risingwavelabs/main-cron/builds/2108#
Importantly the runtime for backfill and e2e test does not regress:
Screenshot 2024-03-21 at 2 52 53 PM

In the main cron build above, the timings are 12m26, 15m40s respectively.

File: Makefile.toml [hunk 1]
```diff
@@ -225,6 +225,17 @@ fi
...
 '''
 
+[tasks.ci-clean-logs]
+private = true
+category = "RiseDev - CI"
+description = "Clean logs before CI"
+script = '''
+#!/usr/bin/env bash
+set -e
+cat /risingwave/.risingwave/log/compactor-6660.log | sed -E 's/table_ids: \[[0-9, ]+\]/table_ids: [hidden]/g' > tmp.log
```
Comment (BugenZhao) [inline Makefile.toml:235]: Is it possible to disable logs for compactor in advance through `RUST_LOG`?

File: src/frontend/planner_test/tests/testdata/output/append_only.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@
     └─StreamProject { exprs: [t1.v1, max(t1.v2)] }
       └─StreamHashAgg [append_only] { group_key: [t1.v1], aggs: [max(t1.v2), count] }
         └─StreamExchange { dist: HashShard(t1.v1) }
-          └─StreamTableScan { table: t1, columns: [t1.v1, t1.v2, t1._row_id], stream_scan_type: Backfill, pk: [t1._row_id], dist: UpstreamHashShard(t1._row_id) }
```
Comment (BugenZhao) [inline src/frontend/planner_test/tests/testdata/output/append_only.yaml:10]: An alternative is to rename...

- `Backfill` to `NoShuffleBackfill`
- `ArrangementBackfill` to `Backfill`

so there's no change on the plans. 😄

File: Makefile.toml (hunk 1 above)
Comment (kwannoel) [reply Makefile.toml:235]: Well I sort of prefer this approach, as it pinpoints and mitigates the exact cause.
If we disable the compactor logs entirely, means we can't check it if there's some error due to compactor.

Comment (BugenZhao) [reply Makefile.toml:235]: No need to disable everything, just the verbose lines. 😄

**Review by chenzl25 — APPROVED** [review]
LGTM

File: Makefile.toml (hunk 1 above)
Comment (kwannoel) [reply Makefile.toml]: Fixed it 👍

**Comment by kwannoel** [comment]
Waiting for main-cron fixes https://github.com/risingwavelabs/risingwave/pull/15861, https://github.com/risingwavelabs/risingwave/pull/15908 before merging this in. To make sure no regressions are caused by arrangement backfill.

**Comment by kwannoel** [comment]
Fix parallel in memory tests here. https://github.com/risingwavelabs/risingwave/pull/15930

**Pull request merged**